src/platform/packages/shared/kbn-scout/src/servers/configs/utils/detect_custom_config.test.ts

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import type { CliSupportedServerModes } from '../../../types';
+import { detectCustomConfigDir, getConfigRootDir } from './detect_custom_config';
+
+describe('detectCustomConfigDir', () => {
+  it('should detect custom config directory from playwright config path', () => {
+    const configPath = 'x-pack/solutions/security/test/scout_uiam_local/ui/playwright.config.ts';
```

Comment:
I concern I have with the `/test/scout_something/ui` folder naming is that a lot of the Scout-specific internals depend on the `/test/scout/ui` or `/test/scout/api` file path. Take a look at the [Scout ESLint rules](https://github.com/elastic/kibana/blob/main/.eslintrc.js#L2697-L2701) for example, these won't work with this custom file path AFAIK. These aren't the only dependency IIRC, but I need to dive deeper to search for more. At the same time, I need to think deep about an alternative way to structure this.

Reply:
there are more places to update ESLint rules:

https://github.com/elastic/kibana/blob/5385f96a132114362b2542e6a44c96a697b66c28/.eslintrc.js#L1594-L1596

https://github.com/elastic/kibana/blob/5385f96a132114362b2542e6a44c96a697b66c28/.eslintrc.js#L2605-L2608

and few more

Reply:
I see we also have a dependency [here](https://github.com/elastic/kibana/blob/5385f96a132114362b2542e6a44c96a697b66c28/src/platform/packages/shared/kbn-scout/src/config/discovery/search_configs.ts#L35-L40) in `search_configs.ts` for config discovery and [here](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-scout/src/common/utils/test_files.ts#L52) for test file path validation, which may be responsible for the error below when running the `node scripts/scout.js run-tests --stateful --testFiles ...` command:

```
ERROR Test file must be within one of /scout/ui/tests, /scout/ui/parallel_tests, /scout/api/tests directories
```

Hold on a moment while I try to think about an alternative file path / structure.

Reply:
I think that's a solid starting point. We can always come back and iterate on this if we need improvements.